internal/cluster/clusters.go

```diff
@@ -100,44 +114,20 @@ func (e *clusterModelNotFoundError) NotFound() bool {
 }
 
 // findOneBy returns a cluster instance for an organization by cluster name.
-func (c *Clusters) findOneBy(organizationID uint, field string, criteria interface{}) (*model.ClusterModel, error) {
-	cluster := model.ClusterModel{
-		OrganizationId: organizationID,
-	}
-
-	switch field {
-	case "id":
-		id, ok := criteria.(uint)
-		if !ok {
-			return nil, errors.New("criteria is not a valid uint value for id")
-		}
-
-		cluster.ID = id
-
-	case "name":
-		name, ok := criteria.(string)
-		if !ok {
-			return nil, errors.New("criteria is not a valid string value for name")
-		}
-
-		cluster.Name = name
-	}
-
-	err := c.db.Where(cluster).Preload("ScaleOptions").First(&cluster).Error
+func (c *Clusters) findOneBy(cluster model.ClusterModel) (*model.ClusterModel, error) {
+	var result model.ClusterModel
+	err := c.db.Where(cluster).Preload("ScaleOptions").First(&result).Error
 	if gorm.IsRecordNotFoundError(err) {
 		return nil, errors.WithStack(&clusterModelNotFoundError{
-			cluster:        criteria,
-			organizationID: organizationID,
+			cluster:        cluster,
+			organizationID: cluster.OrganizationId,
 		})
-	} else if err != nil {
-		return nil, emperror.With(
-			errors.Wrapf(err, "could not get cluster by %s", field),
-			"cluster", criteria,
-			"organization", organizationID,
-		)
+	}
+	if err != nil {
+		return nil, emperror.With(err, "cluster", cluster, "organization", cluster.OrganizationId)
```

Comment:
Check `clusterModelNotFoundError` struct: it already extracts the context from itself, although you might have to update it as well.